netutils/dhcp6c/dhcp6c.c

```diff
@@ -0,0 +1,1951 @@
+    {
+      iov[5].iov_len = sizeof(reconf_accept);
+    }
+  else if (type != DHCPV6_MSG_REQUEST)
+    {
+      if (ia_na_len == 0)
+        {
+          iov[7].iov_len = 0;
+        }
+
+      if (ia_pd_len == 0)
+        {
+          iov[9].iov_len = 0;
+        }
+    }
+
+  if (pdhcp6c->ia_mode == IA_MODE_NONE)
+    {
+      iov[7].iov_len = 0;
+    }
+
+  msg.msg_name = &dest;
+  msg.msg_namelen = sizeof(dest);
+  msg.msg_iov = iov;
+  msg.msg_iovlen = cnt;
+  msg.msg_control = NULL;
+  msg.msg_controllen = 0;
+  msg.msg_flags = 0;
+
+  sendmsg(pdhcp6c->sockfd, &msg, 0);
+}
+
+static int64_t dhcp6c_rand_delay(FAR void *handle, int64_t time)
+{
+  FAR struct dhcp6c_state_s *pdhcp6c = (FAR struct dhcp6c_state_s *)handle;
+  int random;
+
+  read(pdhcp6c->urandom_fd, &random, sizeof(random));
+  return (time * (random % 1000)) / 10000;
+}
+
+static bool dhcp6c_response_is_valid(FAR void *handle, FAR const void *buf,
+                                     ssize_t len,
+                                     const uint8_t transaction[3],
+                                     enum dhcpv6_msg_e type)
+{
+  FAR struct dhcp6c_state_s *pdhcp6c = (FAR struct dhcp6c_state_s *)handle;
+  FAR const struct dhcpv6_header_s *rep = buf;
+  FAR uint8_t *end;
+  FAR uint8_t *odata;
+  uint16_t otype;
+  uint16_t olen;
+  bool clientid_ok = false;
+  bool serverid_ok = false;
+  size_t client_id_len;
+  size_t server_id_len;
+  FAR void *client_id;
+  FAR void *server_id;
+
+  if (len < (ssize_t)sizeof(*rep) ||
+      memcmp(rep->tr_id, transaction, sizeof(rep->tr_id)))
+    {
+      return false;
+    }
+
+  if (type == DHCPV6_MSG_SOLICIT)
+    {
+      if (rep->msg_type != DHCPV6_MSG_ADVERT &&
+          rep->msg_type != DHCPV6_MSG_REPLY)
+        {
+          return false;
+        }
+    }
+  else if (type == DHCPV6_MSG_UNKNOWN)
+    {
+      if (!pdhcp6c->accept_reconfig ||
+          rep->msg_type != DHCPV6_MSG_RECONF)
+        {
+          return false;
+        }
+    }
+  else if (rep->msg_type != DHCPV6_MSG_REPLY)
+    {
+      return false;
+    }
+
+  end = ((FAR uint8_t *)buf) + len;
+  client_id = dhcp6c_get_state(handle, STATE_CLIENT_ID, &client_id_len);
+  server_id = dhcp6c_get_state(handle, STATE_SERVER_ID, &server_id_len);
+
+  dhcpv6_for_each_option(&rep[1], end, otype, olen, odata)
+    {
+      if (otype == DHCPV6_OPT_CLIENTID)
+        {
+          clientid_ok = (olen + 4U == client_id_len) &&
+                        !memcmp(&odata[-4], client_id, client_id_len);
+        }
+      else if (otype == DHCPV6_OPT_SERVERID)
+        {
+          serverid_ok = (olen + 4U == server_id_len) &&
+                        !memcmp(&odata[-4], server_id, server_id_len);
+        }
+    }
+
+  return clientid_ok && (serverid_ok || server_id_len == 0);
+}
+
+static int dhcp6c_command(FAR void *handle, enum dhcpv6_msg_e type)
+{
+  FAR struct dhcp6c_state_s *pdhcp6c = (FAR struct dhcp6c_state_s *)handle;
+  uint8_t buf[1536];
```

Comment:
I suggest to move it to heap and then we can remove `_packed_struct` from related types.